haskell-src/exec/Chainweb/Server.hs

```diff
@@ -442,6 +450,21 @@ toApiTxDetail tx contHist blk evs = TxDetail
     toTxEvent ev =
       TxEvent (_ev_qualName ev) (unPgJsonb $ _ev_params ev)
 
+toApiTxSummary :: Transaction -> ContinuationHistory -> Block -> TxSummary
+toApiTxSummary tx contHist blk = TxSummary
+        { _txSummary_chain = fromIntegral $ _tx_chainId tx
+        , _txSummary_height = fromIntegral $ _block_height blk
+        , _txSummary_blockHash = unDbHash $ unBlockId $ _tx_block tx
+        , _txSummary_creationTime = _tx_creationTime tx
+        , _txSummary_requestKey = unDbHash $ _tx_requestKey tx
+        , _txSummary_sender = _tx_sender tx
+        , _txSummary_code = _tx_code tx
+        , _txSummary_continuation = unPgJsonb <$> _tx_continuation tx
+        , _txSummary_result = maybe TxFailed (const TxSucceeded) $ _tx_goodResult tx
+        , _txSummary_initialCode = chCode contHist
+        , _txSummary_previousSteps = V.toList (chSteps contHist) <$ chCode contHist
+        }
```

Comment:
I don't think you need to write this conversion function. We already have a [`toDbTxSummary`](https://github.com/kadena-io/chainweb-data/blob/f6431722672da3c9414a450fbb0dbebb7d511c7e/haskell-src/lib/ChainwebDb/Queries.hs#L88-L99) function to go from a `Transaction` to a `DbTxSummaryT`. And also a [`dbToApiTxSummary`](https://github.com/kadena-io/chainweb-data/blob/f6431722672da3c9414a450fbb0dbebb7d511c7e/haskell-src/exec/Chainweb/Server.hs#L350-L363) function to convert that to an `TxSummary`. It would be nice to use these functions because those are what the code search handler also uses, so reusing them here would reduce the chance of accidental inconsistencies.

BTW, by using `toDbTxSummary` inside your `queryTxsByPactId` `beam` query, you can also avoid copying unnecessary columns of the `Transaction` row.